src/ast/ast_build_op_contexts.c

```diff
@@ -186,26 +186,42 @@ AST_UnwindContext AST_PrepareUnwindOp(const cypher_astnode_t *unwind_clause) {
 	return ctx;
 }
 
-AST_MergeContext AST_PrepareMergeOp(GraphContext *gc, const cypher_astnode_t *merge_clause,
-									QueryGraph *qg) {
+AST_MergeContext AST_PrepareMergeOp(const cypher_astnode_t *merge_clause, QueryGraph *qg,
+									const char **bound_variables_arr) {
 	const cypher_astnode_t *path = cypher_ast_merge_get_pattern_path(merge_clause);
+	GraphContext *gc = QueryCtx_GetGraphCtx();
+	QueryGraph_AddPath(qg, gc, path); // TODO necessary?
```

Comment:
Might be, test MERGE (a) CREATE  ()-[]->(a) and a like, if this turns out to be mandatory, not sure if this is the right place to introduce this pattern, as this PR removed MERGE clause pattern from QueryGraph construction, so basically the changes has simply migrated the logic.

Reply:
This logic is duplicated from the AST CREATE clause conversion, but I'm not sure if it's required for either clause.